Create Docs for CalDAV

vansia43 opened the pull request:
I did not make this super technical and wrote these up based on some old draft docs we had. For the CalDAV-specific page, I am not sure if we need to go into much more detail. If the service author has any troubleshooting tips or anything like that, I could add it. Otherwise, I think this should be able to help get the user up and running with it.

vansia43 commented:
Not a huge git expert here but I can't seem to figure out why commits from another branch that was merged are on here (even though files changed only shows the files I just changed for this....). Everything is in the last 2: 122a764 and df3e8fd.

Neui commented:
> Not a huge git expert here but I can't seem to figure out why commits from another branch that was merged are on here (even though files changed only shows the files I just changed for this....).

Seems the PR was merged b cherry-picking/rebasing commits on top of master rather than a real merge. So the commits IDs don't match up. I think you should be able to `git fetch --all` to get the newest changes (if you have this upstream repo as a remote), then on the `caldavdocs` branch do `git rebase --onto upstream/master ee180f4` to rebase after commit `ee180f4` onto `upstream/master` (assuming the remote is called `upstream`, might be `origin` on your end).

vansia43 commented:
Thanks. Had to force push, but that worked.

diegogangl commented:
Looks great, there's two things missing I think:

- Tasks need to be tagged `DAV_[CALENDAR_NAME]` to be synced
- The caldav provider might give you a URL for a specific calendar, but actually send the todos from all calendars. The caldav backend will end up syncing all calendars in this case. Nextcloud does this apparently.

Pinging @jaesivsm for more info and a review

Neui reviewed (commented):
The screenshots include "Dev GTG: gtg" window name. You can change it by passing `-t ""` to the `./launch.sh`, so like `./launch.sh -t ""`.

Also, I would have included a page something like "Synchronizing tasks from your calendar service", which a user might be looking for rather than "CalDAV", in case the user doesn't know about CalDAV. I think it would include that the most common option is CalDAV, and just linking to the sync services page and CalDAV page.

vansia43 commented on docs/user_manual/C/gtg-create-sync.page:
```diff
@@ -4,33 +4,36 @@
 
-<p><link href="https://live.gnome.org/gtg/Backends">Sync Services (Backends)</link></p>
+<p><link href="https://github.com/getting-things-gnome/gtg/tree/64d535aee2c59e78fdc269ddd4baaecf257b64b4/GTG/backends"><app>GTG</app> Backends</link></p>
 
-<p><link href="https://live.gnome.org/gtg/dbus">DBus</link></p>
+<p><link href="https://github.com/getting-things-gnome/gtg/blob/master/docs/contributors/DBus%20API%20v1.md">DBus</link></p>
```
So I wonder if it is even worth having this page. The old wiki links were dead, and that was the best I could find. I think I’ll remove it.

Neui replied:
I'd be OK with the page being removed. Like 99,9% of users won't even care about this since it requires programming knowledge and effort. No need to allocate an user-facing page for it. At most it could be in the Contributing to GTG help page (maybe something like stuff that can be done for the project, like "adding new sync services" or whatever), but I wouldn't do it in this PR specifically.

vansia43 commented on docs/user_manual/C/gtg-add-sync.page:
```diff
@@ -6,44 +6,44 @@
+  
+   <p>To enable or disable Sync Services, from the <app>GTG</app> main menu:</p>
+<steps>
+    <item><p>Select <em style="strong">Synchronization.</em></p></item>
```
Good catch. Done.

vansia43 commented on docs/user_manual/C/gtg-caldav.page:
```diff
@@ -0,0 +1,57 @@
+
+  <title>CalDAV Sync Service</title>
+
+  <p>CalDAV is a calendaring protocol that allows a client to access items from a server. Sync your third-party applications that support the CalDAV standard with <app>GTG</app> to manage all of your tasks and calendar items in one place. The options and required fields for the CalDAV Sync Service are described below. Once you have filled in all of the required information, click <em style="stron ...
```
Done.

vansia43 commented on docs/user_manual/C/gtg-sync.page:
```diff
@@ -6,56 +6,44 @@
 
-<p><em style="strong">Evolution tasks</em></p>
-<p>Allows you to synchronize your GTG tasks with Evolution tasks (full synchronization). <em>This sync service is currently disabled because of a bug between PyGTK and python-evolution.</em></p>
+<p><em style="strong">CalDAV</em></p>
```
Done. There was already a "see also" link that gets added to the bottom for this, but I also added it in the list, so they can't miss it.

vansia43 commented:
Thanks, @diegogangl  ... added these two:
> * Tasks need to be tagged `DAV_[CALENDAR_NAME]` to be synced
> * The caldav provider might give you a URL for a specific calendar, but actually, send the todos from all calendars. The caldav backend will end up syncing all calendars in this case. Nextcloud does this apparently.

@Neui  Thanks for the thorough review. I have addressed most of your comments. See my thoughts on the `create-sync` page. I think we could just remove it if we do not have good/stable links to put here. That was the best I could find. 

> Also, I would have included a page something like "Synchronizing tasks from your calendar service", which a user might be looking for rather than "CalDAV", in case the user doesn't know about CalDAV. I think it would include that the most common option is CalDAV, and just linking to the sync services page and CalDAV page.

Instead of adding another page, I have renamed the CalDAV page to **CalDAV: Calendar Sync Service**. The user will see this directly on the `index` page. They can also search in Yelp, so it'll come right up. In the future, if we get more sync services, I am not sure if I would keep them on the `index` because it will get cluttered, but I think it's fine there for now.

Neui commented on docs/user_manual/C/gtg-caldav.page:
```diff
@@ -26,9 +26,11 @@
+  <title>CalDAV: Calendar Sync Service</title>
 
-  <p>CalDAV is a calendaring protocol that allows a client to access items from a server. Sync your third-party applications that support the CalDAV standard with <app>GTG</app> to manage all of your tasks and calendar items in one place. The options and required fields for the CalDAV Sync Service are described below. Once you have filled in all of the required information, click <em style="stron ...
+  <p>CalDAV is a calendaring protocol that allows a client to access items from a server. CalDAV is often provided by calendar and task/groupware services. Sync your third-party applications that support the CalDAV standard with <app>GTG</app> to manage all of your tasks and calendar items in one place. The options and required fields for the CalDAV Sync Service are described below. Once you have ...
```
I think the wording got a bit weird here with the groupware addition, like you say it is often provided by groupware services, and then "Sync your third-party applications that support the CalDAV standard with GTG to manage all of your tasks and calendar items in one place.". 

My try:

```suggestion
  <p>CalDAV is a calendaring protocol that allows a client to access items from a server. Many calendar and task services (often included in groupware packages) provide this feature. <app>GTG</app> supports for this standard to be able to manage all your tasks in one place. The options and required fields for the CalDAV Sync Service are described below. Once you have filled in all of the required information, click <em style="strong">Enable syncing</em> to start importing your CalDAV items.</p> 
```

Neui commented:
> Instead of adding another page, I have renamed the CalDAV page to CalDAV: Calendar Sync Service

OK.

Also I've noticed that the title don't use the -ing suffix for the first verb ("Understand Synchronization Services" vs "Understand*ing* Synchronization Services") like other pages do. I think it also sounds a bit weird, like a command. I'd change it to match the other pages (so adding the -ing).

vansia43 commented:
> Also I've noticed that the title don't use the -ing suffix for the first verb ("Understand Synchronization Services" vs "Understand_ing_ Synchronization Services") like other pages do.

Yea, I missed that. I prefer gerund to imperative as well. Updated for both titles with verbs.

Added the tip regarding the module. I do want to point out again that the `caldav` module is noted in the bulleted list on the Dependencies section in the README, but I don't see it in the code block below that. Not sure if it needs to go there. In any case, I did give the `pip` installation command in the tip I added on the `caldav` page for just that module so that the user isn't stuck and doesn't have to go back and search the README.

diegogangl commented:
>  See my thoughts on the create-sync page. I think we could just remove it if we do not have good/stable links to put here. That was the best I could find.

Agree on this. This is more developer-oriented and better served by having something in the docs folder (and/or asking in an issue here)

> the caldav module is noted in the bulleted list on the Dependencies section in the README, but I don't see it in the code block below that. 

Thanks, added the apt command. It's not available in Fedora though

diegogangl commented:
Merged, thanks!
